back/api/models.py

```diff
@@ -2,10 +2,50 @@
 
 # Create your models here.
 
+"""Lina s'occupe de l'authentification et des classes Student et Teacher
+ici Member peut être un Student ou un Teacher"""
+
 class Member(models.Model):
     username = models.CharField(null=True, blank=True, max_length = 255)
     updated = models.DateTimeField(auto_now=True)
     created = models.DateTimeField(auto_now_add=True)
 
-    def __str__(self):
-        return self.username[0:50]
\ No newline at end of file
+    def __str__(self) -> str:
+        return self.username[0:50]
+
+class Exercise(models.Model):
+    """ Exerices composé de l'énoncé, de la solution écrite par le professeur,
+    des test à lancer (défini par le professeur) et des output de sortie qu'il
+    doit ressortir (pas forcément défini par le professeur)"""
+    subject = models.CharField(null=True, max_length = 255)
+    solution = models.CharField(null=True, max_length = 255)
+    test_input = models.CharField(null=True, max_length = 255)
+    correct_output = models.CharField(null=True, max_length = 255)
+
+    def __str__(self) -> str:
+        return self.subject
+
+class Classroom(models.Model):
+    """ Une classroom est identifiée par sa clé primaire (room_id) et de l'url
+    vers la salle (room_url). Une classe est crée par un Teacher et rejoint par
+    plusieurs students. Elle est aussi composée de plusieurs exercices."""
+    room_id = models.IntegerField(null=True, default=None)
+    room_url = models.URLField(null=True, default=None)
+    teacher = models.OneToOneField(Member, on_delete=models.PROTECT, default=None, related_name="teacher_admin")
+    students = models.ForeignKey(Member, on_delete=models.CASCADE, default=[], related_name="students_on_classroom")
+    subjects = models.ForeignKey(Exercise, on_delete=models.PROTECT, default=[], related_name="subjects_of_classroom")
```

Comment:
I suggest to remove `room_url`, it will be automatically computed with the room id.
For `teacher`, a models.OneToOneField cannot be suitable because it would mean that a teacher would have only one classroom and linked forevet to it.
`students` is at the wrong place ! Written like that, a classroom can take a single student. It will be better to assign to each student a classroom.
Same for `subjects`, written like that, you can only take one subject. Let think about another model field.

Reply:
I understand your point. I let the ForeignKey students in Classroom but I am moving the teacher ForeignKey into the Teacher class. I am also not changing the ForeingKey sujects.

Warning : ForeignKey written like that is not a OneToOne Field, is by default a OneToMany Field. To change the Foreign Key in a OneToOne Field, I would write inside exclusive = True.